addons/mail/static/src/models/thread/thread.js

```diff
@@ -1192,6 +1262,26 @@ function factory(dependencies) {
             return [['replace', this.messages.filter(message => message.isNeedaction)]];
         }
 
+        /**
+         * @returns {integer|undefined}
+         * @private
+         */
+        _computeMessageBeforeNewMessageSeparatorId() {
+            if (this.model !== 'mail.channel') {
+                return clear();
+            }
+            if (this.localMessageUnreadCounter === 0) {
+                return clear();
+            }
+            if (!this.lastSeenByCurrentPartnerMessageId) {
+                return this.lastTransientMessage ? this.lastTransientMessage.id : 0;
+            }
+            if (!this.lastTransientMessage) {
+                return this.lastSeenByCurrentPartnerMessageId;
+            }
+            return Math.max(this.lastTransientMessage.id, this.lastSeenByCurrentPartnerMessageId);
```

Comment:
I don't know, for my part I considered all transient messages like it was "authored by current user", I mean, it's him that request those messages so it should never be considered as not seen I think.

Reply:
For me the rule should be:

If the thread is all seen, and if transient messages are at the bottom, they should be considered seen (because there is no way to mark them as seen otherwise).

If transient messages are before the last seen message, they should be considered seen as well (they are before the last seen message, this is a normal case for all messages).

But if transient messages after somehow after the last seen message in the thread, they should be considered as unseen and the counter adapted accordingly. This is necessary for the consistency of the UI: if there is a separator on a real message, and there are 4 messages including 1 transient after it, counter should be 4 and not 3.

Reply:
Consider the following situation :
```
[message 1]
[message 2]
[message 3] <-- lastSeenMessage
[message 4]
[message 5 (transient)]
[message 6 (transient)]
```
--> unread counter should be 3, right ?

then :
```
lastSeenMessage is false.
[message 1]
[message 2]
[message 3]
[message 4]
[message 5 (transient)]
[message 6 (transient)]
```
unread counter should be 6 or 4 ?

and finally

lastSeenMessage is false.
```
[message 1 (transient)]
[message 2 (transient)]
```
unread counter should be 2 or 0 ?

Reply:
Case 1: 3 (we can see 3 messages after separator)
Case 2: 6 (we can see 6 messages after the separator)
Case 3: 0 (there is no separator, we have to ignore transient in the count, because there would be no real message to "mark them as read" so they would stay unread forever in that case)

(Case 1b) Just make sure also that
```
[message 1]
[message 2 (transient)]
[message 3]
[message 4] <-- lastSeenMessage
[message 5]
[message 6 (transient)]
[message 7 (transient)]
```
Is also 3 and not 4 (transient before separator not counted)

Reply:
Btw in case 1, if the last seen message was 4 instead of 3, we would be in the same situation as case 3 and the count should be 0.

Reply:
It's really complex... but if there is no other way of doing..

Reply:
We might be able to simplify the conditions by considering `lastSeenMessage` `false` exactly the same as if it was `0`.

And then the condition is just about locally counting the number of messages we have after `lastSeenMessage`, just like it was before (not filtered on transient).

And the only complication is when `lastSeenMessage` = "last real message" we just ignore all transient that are after it